src/services/music-providers/youtube-music/youtube-music.service.ts

```diff
@@ -0,0 +1,286 @@
+            playlist.id,
+            { limit: this.getPlaylistTracksLimit },
+        );
+
+        const trackWithId = tracks.filter((t) => {
+            if (t?.id) {
+                return true;
+            }
+
+            this.logService.warn(
+                `Track ${t.name} by ${t.artists} was not removed in youtube. Reason id is absent`,
+            );
+        });
+
+        const trackFromServer = playlistFromServer.tracks.filter(
+            (trackFromServer) =>
+                trackWithId.some(
+                    (propsTrack) => propsTrack.id === trackFromServer.videoId,
+                ),
+        );
+
+        const trackFromServerWithSetVideoId = trackFromServer.filter((t) => {
+            if (t.setVideoId) {
+                return true;
+            }
+
+            this.logService.warn(
+                `Track ${t.title} by ${t.artists} was not removed in youtube. Reason setVideoId is absent`,
+            );
+        });
+
+        if (!trackFromServerWithSetVideoId.length) {
+            return;
+        }
+
+        await this.removePlaylistItemsWrapperOverLibmuseMethod(
+            playlist.id,
+            trackFromServerWithSetVideoId.map((t) => ({
+                videoId: t.videoId,
+                setVideoId: t.setVideoId as string,
+            })),
+        );
+    }
+
+    private artistsToNameArray(artists: SongArtist[]) {
+        let result: string[] = [];
+
+        for (const artist of artists) {
+            if (artist.id) {
+                result.push(artist.name);
+                continue;
+            }
+
+            const names = artist.name.replace(/, & | & | and | и /g, ', ');
+            result = [...result, ...names.split(', ')];
+        }
+
+        return result;
+    }
+
+    private checkThatTrackIsSuitable(
+        findName: string,
+        findArtist: string[],
+        foundName: string,
+        foundArtists: string[],
+    ) {
+        findName = findName.toLowerCase();
+        findArtist = findArtist.map((a) => a.toLowerCase());
+        foundName = foundName.toLowerCase();
+        foundArtists = foundArtists.map((a) => a.toLowerCase());
+
+        const findNameWithoutParticipantDesignations =
+            this.getTrackNameWithoutParticipantDesignations(findName);
+        const foundNameWithoutParticipantDesignations =
+            this.getTrackNameWithoutParticipantDesignations(foundName);
+
+        if (
+            findNameWithoutParticipantDesignations !==
+            foundNameWithoutParticipantDesignations
+        ) {
+            return false;
+        }
+
+        const featuringArtists = findArtist.filter(
+            (a) => foundName.indexOf(a) !== -1,
+        );
+
+        const allFoundArtists = [
+            ...new Set([...foundArtists, ...featuringArtists]),
+        ];
+        return checkIfArraysAreEqual<string>(findArtist, allFoundArtists);
+    }
+
+    private getTrackNameWithoutParticipantDesignations(name: string): string {
```

Comment:
The function name `getTrackNameWithoutParticipantDesignations` seems to be verbose and lacks precision. Consider renaming it to `extractTrackName` or `getPrimaryTrackName` for clarity and simplicity, improving readability and maintainability of the code.